Revise the CSSOM entry in the Glossary

sideshowbarker opened the pull request:
This updates the CSSOM·entry·in·the·Glossary to more closely match what we have at https://developer.mozilla.org/en-US/docs/Web/API/CSS_Object_Model.  Otherwise, without this change, what we have in the Glossary for CSSOM isn’t really an improvement over what we have at API/CSS_Object_Model — and also contains grammar and style problems that would otherwise need fixing.

Fixes https://github.com/mdn/content/issues/6146

sideshowbarker committed:
Revise the CSSOM entry in the Glossary

This updates the CSSOM·entry·in·the·Glossary to more closely match what we
have at https://developer.mozilla.org/en-US/docs/Web/API/CSS_Object_Model.
Otherwise, without this change, what we have in the Glossary for CSSOM isn’t
really an improvement over what we have at API/CSS_Object_Model — and also
contains grammar and style problems that would otherwise need fixing.

Fixes https://github.com/mdn/content/issues/6146

terrifricker commented:
The wording in the CSSDOM glossary was much more informative to me than what is in the CSSDOM API glossary entry.  I have added the sentence I mentioned to my notes on the subject. I actually took time to copy and paste it because it was so enlightening on the subject.

I strongly suggest only removing the one word rather than the whole top section, removing the most important section of the glossary entry.

terrifricker commented on files/en-us/glossary/cssom/index.html:
```diff
@@ -6,15 +6,8 @@
   - Glossary
-  - Web Performance
 ---
-<p>The <strong>CSS Object Model (CSSOM)</strong> is a map of all CSS selectors and relevant properties for each selector in the form of a tree, with a root node, sibling, descendant, child, and other relationship. The CSSOM is very similar to the {{glossary('DOM', 'Document Object Model (DOM)')}}. Both of them are part of the critical rendering path which is a series of steps that must happen to  ...
```
This paragraph is too useful to delete

sideshowbarker replied:
>> `<p>The <strong>CSS Object Model (CSSOM)</strong> is a map of all CSS selectors and relevant properties for each selector in the form of a tree, with a root node, sibling, descendant, child, and other relationship. The CSSOM is very similar to the {{glossary('DOM', 'Document Object Model (DOM)')}}. Both of them are part of the critical rendering path which is a series of steps that must happen to properly render a website.</p>`
>
> This paragraph is too useful to delete

Can you explain exactly what parts of it seem useful?

The CSSOM is in fact not a “a map of all CSS selectors and relevant properties for each selector”.

That statement doesn’t even make sense. The CSSOM is not a map. And selectors do not have properties.

And the the CSSOM is “a tree, with a root node”, could you explain what exactly what you understand the root node of the CSSOM tree to be? — because I definitely have no idea what that root node might be, and there’s nothing in the CSSOM spec that talks about such a root node; nor is there anything in the CSSOM spec that talks all all about the CSSOM even being a tree.

And in the mention of “sibling, descendant, child, and other relationship”, can you explain exactly what your understand “other relationship” to be referring to? — because I have no idea what it might be referring to. The CSSOM itself anyway doesn’t have anything to do with “sibling, descendant, child” — instead whoever wrote that seems to have been thinking of the DOM, and about maybe how selectors select parts of the DOM. But regardless, that’s not what the CSSOM is.

terrifricker commented on files/en-us/glossary/cssom/index.html:
```diff
@@ -6,15 +6,8 @@
 ---
-<p>The <strong>CSS Object Model (CSSOM)</strong> is a map of all CSS selectors and relevant properties for each selector in the form of a tree, with a root node, sibling, descendant, child, and other relationship. The CSSOM is very similar to the {{glossary('DOM', 'Document Object Model (DOM)')}}. Both of them are part of the critical rendering path which is a series of steps that must happen to  ...
-
-<p>The CSSOM, together with the DOM, to build the render tree, which is in turn used by the browser to layout and paint the web page.</p>
```
This paragraph is too useful to delete.  Only the word 'to' (the seventh word) should be removed to correct the grammar.

sideshowbarker replied:
>> ` <p>The CSSOM, together with the DOM, to build the render tree, which is in turn used by the browser to layout and paint the web page.</p>`
> 
> This paragraph is too useful to delete. Only the word 'to' (the seventh word) should be removed to correct the grammar.

To repeat for the record here what I wrote over at https://github.com/mdn/content/issues/6146#issuecomment-864627596:

> The CSSOM and the DOM do not “build the render tree”. The _browser_ engine builds the render tree. The CSSOM does no building of anything, and the DOM does no building of anything. Instead, the _browser_ builds the CSSOM and the browser builds the DOM. And then the built DOM has APIs that expose methods and properties you can use for reading and modifying the document’s structure and content, and the built CSSOM has APIs that expose methods and properties you can use for reading and modifying the document’s style information.
> 
> But that’s not at all what’s said in the existing paragraphs that I replaced. Those paragraphs seem to have been written by someone who didn’t actually understand what the CSSOM is, nor what the DOM is. What they seem to have been trying to describe is something quite different from what the CSSOM actually is.

terrifricker replied:
I am very glad to know that the glossary will no longer cause misunderstanding.

terrifricker reviewed (changes requested):
This pull request goes too far and removes information that clearly describes what the CSSDOM is and how it works with the DOM.  The CSSDOM is actually different from its API, thus the descriptions should be different.

sideshowbarker commented:
> This pull request goes too far and removes information that clearly describes what the CSSDOM is and how it works with the DOM.

In fact this pull request doesn’t go too far and the information I removed that seemed to be attempting to describe what the CSSOM is and how it works with the DOM was very poorly written and was actually wrong in a number of ways. The parts I removed do not at all align with what the CSSDOM spec defines the CSSOM to be, and does not align with how the CSSOM is implemented in browser engines.

> The CSSDOM is actually different from its API, thus the descriptions should be different.

Can you explain in your own words exactly how the CSSOM is different from its API? — because, having read the CSSOM spec quite a bit, and having some understanding how of the CSSOM is implemented in browsers, I don’t understand how I would describe the CSSOM without explaining it in terms of its API.

The only thing that’s more detailed that I can imagine to say is that the CSSOM relies on an in-memory representation of a document’s style information — similar to how the DOM relies on an in-memory representation of the document’s structure and content. But as a web developer, the only way I can programmatically observe and interact with the CSSOM’s in-memory representation is by using the CSSOM API that the browser engine exposes for me — just as the only way I can observe and interact with the in-memory representation of the document’s structure and content is by using the DOM API that the browser exposes.

But the parts I removed in this pull request didn’t say anything like that. And anyway, for all practical purposes, the CSSOM isn’t different from its API, and the DOM isn’t different from its API. In other words, the CSSOM and the DOM aren’t the in-memory representations that the browser uses internally — they are instead the representation that the CSSOM API and DOM API expose to JavaScript.

sideshowbarker commented:
> The wording in the CSSDOM glossary was much more informative to me than what is in the CSSDOM API glossary entry. I have added the sentence I mentioned to my notes on the subject. I actually took time to copy and paste it because it was so enlightening on the subject.

With respect, if you believe that the CSSOM is what that existing glossary entry seems to be trying to describe it as, then you have a misunderstanding about what the CSSOM actually is. To get a better understanding, I’d suggest you consider reading the CSSOM spec itself at https://drafts.csswg.org/cssom/. I think you’ll find it’s not at all like what the existing glossary entry is trying to describe it as.

It’s not good for MDN to be causing misunderstandings like this. It’s not good for future MDN readers to read that glossary entry and go away thinking they’ve gained an understanding of the CSSOM when in fact what’s happened is that MDN has caused to them seriously misunderstand it.

It’s important for everybody reading MDN to understand that not everything in MDN is necessarily correct at all, and not everything in MDN has necessarily even ever been reviewed by people who understand the subject matter.

In the years before we moved the MDN sources to GitHub and put in place a review process for all changes, people could just come in and add and change MDN content however they wanted to — without anybody else reviewing it at all for accuracy.

So there are literally dozens or even hundreds of places where there’s very wrong information in MDN that somebody years ago wandered in and added without anybody else ever checking it. In particular there are numerous such cases in the Glossary. The main way we get those fixed is when somebody reports some problem with the content — as in this case, where the one grammar problem was reported — and then somebody else goes in and looks at the content and realizes how bad it is, and how badly it needs to be replaced with something that actually makes sense and that’s actually accurate.

This is one of those cases.

terrifricker reviewed (approved):
I'd like to remove myself from the review process.

sideshowbarker commented:
Hi Terri,

> I'd like to remove myself from the review process.

There’s an <kbd>**Unsubscribe**</kbd> button you can use:

![Untitled-1](https://user-images.githubusercontent.com/194984/122866130-6fbf5d00-d362-11eb-911c-383e2ff083bc.png)

Rumyra reviewed (approved):
Looks good to me @sideshowbarker
